crates/exex/exex/src/manager.rs

```diff
@@ -481,18 +481,190 @@ impl Clone for ExExManagerHandle {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use reth_primitives::{SealedBlockWithSenders, B256};
+    use reth_provider::Chain;
+
+    #[tokio::test]
+    async fn test_delivers_events() {
+        let (mut exex_handle, event_tx, mut _notification_rx) =
+            ExExHandle::new("test_exex".to_string());
+
+        // Send an event and check that it's delivered correctly
+        event_tx.send(ExExEvent::FinishedHeight(42)).unwrap();
+        let received_event = exex_handle.receiver.recv().await.unwrap();
+        assert_eq!(received_event, ExExEvent::FinishedHeight(42));
+    }
+
     #[tokio::test]
-    async fn delivers_events() {}
+    async fn test_updates_block_height() {
+        let (mut exex_handle, event_tx, mut _notification_rx) =
+            ExExHandle::new("test_exex".to_string());
+
+        // Check initial block height
+        assert!(exex_handle.finished_height.is_none());
+
+        // Update the block height via an event
+        event_tx.send(ExExEvent::FinishedHeight(42)).unwrap();
+        let received_event = exex_handle.receiver.recv().await.unwrap();
+        assert_eq!(received_event, ExExEvent::FinishedHeight(42));
+
+        // Check that the block height was updated
+        exex_handle.finished_height = Some(42);
+        assert_eq!(exex_handle.finished_height, Some(42));
```

Comment:
i don't understand this part, aren't we updating the field by ourself and then it's obviously always equal to the value that we set it to?

Reply:
Yes I forgot to update what I wanted to test: update the finished height via an event. Just fixed.